src/scriptdev2/scripts/eastern_kingdoms/shadowfang_keep/shadowfang_keep.cpp

```diff
@@ -598,8 +598,11 @@ struct boss_arugalAI : public ScriptedAI
         {
             if (!m_bAttacking)
             {
-                DoCastSpellIfCan(m_creature->getVictim(), SPELL_VOID_BOLT);
-                m_uiVoidboltTimer = urand(2900, 4800);
+                if (Unit* victim = m_creature->getVictim())
```

Comment:
@xfurry But on line 564 in this script we do a 'StopAttacking()' - doesn't that nullify the victim? Then we would in fact have to recheck it here.

Shouldn't it be a ((Unit*)this)->AttackStop(); if we indeed want to keep the victim but stop meleeing?